### PR TITLE
[testing] [CI] [PA] support Protected Apps in CI for tests

### PR DESCRIPTION
**Pull request by mokshjawa**
Merge after #97 

Tested here: https://github.com/tryretool/admin-protected-apps/actions (actions are running on `moksh/pa-ci`)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ services:
       - ./keys:/root/.ssh
       - ssh:/retool_backend/autogen_ssh_keys
       - ./retool:/usr/local/retool-git-repo
+      - ${BOOTSTRAP_SOURCE}:/usr/local/retool-repo
```
Comment (mokshjawa) [inline docker-compose.yml:34]: also covering this https://github.com/tryretool/retool-onpremise/pull/96 (so whatever we merge first)

File: testing/generate.js [hunk 2]
```diff
@@ -210,17 +213,16 @@ function main() {
...
   }
   
-  // TODO: Support protected applications
   const protectedPath = path.join(basePath, '.retool', 'protected-apps.yaml');
-  if (fs.existsSync(protectedPath)) {
-    console.log('Testing Protected Applications in CI is currently not supported');
-    process.exit(1);
+  if (!fs.existsSync(protectedPath)) {
+    console.log('Protected apps repository is missing ./retool/protected-apps.yaml');
+    process.exit(0);
```
Comment (avimoondra) [inline testing/generate.js:219]: @mokshjawa will need to update for Gitlab (file does not exist)

File: testing/generate.js [hunk 3]
```diff
@@ -241,21 +243,26 @@ function main() {
...
-
-      if (parsed.dir !== path.join(basePath, 'apps')) {
-        const dirName = path.basename(parsed.dir);
-
-        if (dirName !== 'archive') {
-           const fileName = slugify(`${dirName}-${parsed.name}`, {lower: true});
+      const parsed = path.parse(file);
+      const parentDirectory = parsed.dir
+      const grandparentDirectory = path.parse(parsed.dir).dir 
+      const directory = usingProtectedApps ? grandparentDirectory : parentDirectory
```
Comment (avimoondra) [inline testing/generate.js:251]: will break with arbitrary folder nesting - but can log that for later!

**Review by avimoondra — APPROVED** [review]
Conditional on some name changes! 🙏

File: testing/generate.js (hunk 2 above)
Comment (mokshjawa) [reply testing/generate.js]: https://linear.app/retool/issue/DTC-2230/protected-apps-check-in-ci-will-not-work-for-gitlab

**Pull request merged**